### DISCUSSION TITLE
OSX Silicon M1 installation

### DISCUSSION THREAD

**Discussion by GillesDuvert** [gnudatalanguage/gdl#1675] [General]
I've made some tests on an OSX M1, Sonoma 14.0 .
There is absolutely NO WAY for me to convince the mac to run gdl except by doing this (below). In particular there is no hope to get a DMG file for easy install on other M1 machines. I will amend the Wiki documentation accordingly.
Now, for the method:

- use exclusively build_gdl.sh, last version
- do `build_gdl.sh prep` , then "`build_gdl.sh configure`".
- At this point, go into `gdl/build` and type:
- `cmake -DCMAKE_INSTALL_PREFIX="/usr/local" -DOPENMP=OFF .. `
- `make -j 8`  (8 is an example of number of threads)
- sudo make install
- and voila gdl is installed

No other possibility found after hours of struggle.  Installing with openmp make gdl crash; installing elsewhere than /usr/local (using, e.g. "build_gdl.sh install" amounts to the same. Not mentioning making an installer with build_gdl pack + build_gdl prep_deploy (that creates a dmg file), this will not work at all.

There are probably other ways. Getting the multithreading  working would be great.

**Comment by GillesDuvert** [comment]
Also, the known problems (slowness of loops eg in time_test4) are present.

**Comment by jaymurthy** [comment]
Hello Gilles,
I'm missing something. I started in /opt/homebrew/gdl. 
build_sh prep worked fine but now I'm getting an error from build_sh configure:
./build_gdl.sh configure
[build_gdl.sh] Configuring GDL (Release)...
CMake Warning:
  Ignoring extra path from command line:

   "/opt/homebrew/gdl/.."


CMake Error: The source directory "/opt/homebrew" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.   17h21m  
$ pwd
/opt/homebrew/gdl

**Comment by GillesDuvert** [comment]
Hello @jaymurthy , I did not try compiling gdl elsewhere than in my $HOME directory. So, git clone created a 'gdl' directory there and every command described above is run with `$HOME/gdl/scripts/build_gdl.sh` and refer to the directories in `$HOME/gdl`, mainly `$HOME/gdl/build` where the `cmake` command must be called.
When in `$HOME/gdl/build`, the command `cmake -DCMAKE_INSTALL_PREFIX="/usr/local" -DOPENMP=OFF ..` contains `..` since `..` seen from there is the (parent) directory with the relevant `CMakeLists.txt`. You may have foud a restriction in the use of build_gdl.sh, that it cannot be elsewhere than `$HOME/gdl/scripts/build_gdl.sh`. Please try to do everything (git clone, build_gdl.sh...) from your login directory, not /opt/homebrew.

**Reply by jaymurthy** [reply]
Thank you Gilles. That worked perfectly. 
Jayant
